injector/src/main/java/edu/ucr/cs/riple/injector/changes/ChangeVisitor.java

```diff
@@ -0,0 +1,183 @@
+    final AtomicReference<Modification> ans = new AtomicReference<>();
+    final NodeList<BodyDeclaration<?>> members = pair.a;
+    final Change change = pair.b;
+    members.forEach(
+        bodyDeclaration ->
+            bodyDeclaration.ifFieldDeclaration(
+                fieldDeclaration -> {
+                  if (ans.get() != null) {
+                    // already found the member.
+                    return;
+                  }
+                  NodeList<VariableDeclarator> vars =
+                      fieldDeclaration.asFieldDeclaration().getVariables();
+                  for (VariableDeclarator v : vars) {
+                    if (onField.variables.contains(v.getName().toString())) {
+                      ans.set(change.visit(fieldDeclaration));
+                      break;
+                    }
+                  }
+                }));
+    return ans.get();
+  }
+
+  @Override
+  @Nullable
+  public Modification visitParameter(
+      OnParameter onParameter, Pair<NodeList<BodyDeclaration<?>>, Change> pair) {
+    final AtomicReference<Modification> ans = new AtomicReference<>();
+    final NodeList<BodyDeclaration<?>> members = pair.a;
+    final Change change = pair.b;
+    members.forEach(
+        bodyDeclaration ->
+            bodyDeclaration.ifCallableDeclaration(
+                callableDeclaration -> {
+                  if (ans.get() != null) {
+                    // already found the member.
+                    return;
+                  }
+                  if (onParameter.matchesCallableDeclaration(callableDeclaration)) {
+                    NodeList<?> params = callableDeclaration.getParameters();
+                    if (onParameter.index < params.size()) {
+                      if (params.get(onParameter.index) != null) {
+                        Node param = params.get(onParameter.index);
+                        if (param instanceof Parameter) {
+                          ans.set(change.visit((Parameter) param));
+                        }
+                      }
+                    }
+                  }
+                }));
+    return ans.get();
+  }
+
+  @Override
+  @Nullable
+  public Modification visitClass(OnClass onClass, Pair<NodeList<BodyDeclaration<?>>, Change> pair) {
+    final NodeList<BodyDeclaration<?>> members = pair.a;
+    final Change change = pair.b;
+    if (isAnonymousClassFlatName(change.location.clazz)) {
+      return null;
+    }
+    // Get the enclosing class of the members
+    Optional<Node> optionalClass = members.getParentNode();
+    if (optionalClass.isEmpty() || !(optionalClass.get() instanceof BodyDeclaration<?>)) {
+      return null;
+    }
+    return change.visit(((BodyDeclaration<?>) optionalClass.get()));
+  }
+
+  /**
+   * Applies the change to the compilation unit.
+   *
+   * @param change the change to apply.
+   * @return the modification that should be applied.
+   */
+  @Nullable
+  public Modification visit(Change change) {
```

Comment:
It is confusing to name this method `visit` since it's not one of the standard visitor methods.  How about:
```suggestion
  public Modification computeModification(Change change) {
```

Reply:
Sure, that makes sense. 15ff010019a8ee78522ed2f7905546c62f94be8d